frontend/dockerfile/instructions/parse.go

```diff
@@ -80,6 +80,9 @@ func ParseInstructionWithLinter(node *parser.Node, lintWarn linter.LintWarnFunc)
 	case command.Env:
 		return parseEnv(req)
 	case command.Maintainer:
+		if lintWarn != nil {
```

Comment:
I wonder if we could just have this nil check inside the `Run()` one time so every caller does not need to check it.